File: app/src/main/java/sk/styk/martin/apkanalyzer/adapter/detaillist/GenericDetailListAdapter.java
package sk.styk.martin.apkanalyzer.adapter.detaillist;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

import java.util.List;

/**
 * @author Martin Styk
 * @version 12.10.2017.
 */
public abstract class GenericDetailListAdapter<T, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {

    private final List<T> items;

    public GenericDetailListAdapter(@NonNull List<T> items) {
        super();
        setHasStableIds(true);
        this.items = items;
    }

    @Override
    public abstract void onBindViewHolder(final VH holder, final int position);

    @Override
    public int getItemCount() {
        return items.size();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public T getItem(int position) {
        return items.get(position);
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

}

File: app/src/main/java/sk/styk/martin/apkanalyzer/adapter/detaillist/GenericDetailListAdapter.kt
package sk.styk.martin.apkanalyzer.adapter.detaillist

import android.support.v7.widget.RecyclerView

/**
 * @author Martin Styk
 * @version 12.10.2017.
 */
abstract class GenericDetailListAdapter<T, VH : RecyclerView.ViewHolder>(private val items: List<T>) : RecyclerView.Adapter<VH>() {

    init {
        this.setHasStableIds(true)
    }

    abstract override fun onBindViewHolder(holder: VH, position: Int)

    override fun getItemCount(): Int = items.size

    override fun getItemId(position: Int): Long = position.toLong()

    fun getItem(position: Int): T = items[position]

    override fun getItemViewType(position: Int): Int = position
}
